[docs] Streamline beats getting started guides

dedemorton opened the pull request:
This PR moves extraneous content out of the Beats "getting started" topics and adds a tabbed widget for displaying platform-specific info (see NOTES below).

Preview: https://beats_17007.docs-preview.app.elstc.co/diff

This is a very big PR. To make it easier for you, I've collected a list of topics that you should view in the HTML preview (rather than the diff). Each team should review topics for the areas they own. Also look at Files Changed for other small changes that affect your area.

**Rules of engagement:**

* Please check off the item below and add your initials when you are done reviewing the topic. You can skip reviewing files that have Javascript and HTML--we have someone lined up to help us improve the code.
* I still think that some of our command examples that show sudo could be wrong. Please add a comment if you something that's wrong.

**AUDITBEAT**

- [x] [Quick start](https://beats_17007.docs-preview.app.elstc.co/guide/en/beats/auditbeat/master/auditbeat-installation-configuration.html) | [Diff](https://github.com/elastic/beats/pull/17007/files#diff-a18455b26269a49e04471eee89f30100)

**FILEBEAT**
- [x] [Configure modules](https://beats_17007.docs-preview.app.elstc.co/guide/en/beats/filebeat/master/configuration-filebeat-modules.html) | [Diff](https://github.com/elastic/beats/pull/17007/files?#diff-3a37a0837b92ad36e51f3e8a460d3eb5)
- [x] [Configure inputs](https://beats_17007.docs-preview.app.elstc.co/guide/en/beats/filebeat/master/configuration-filebeat-options.html) | [Diff](https://github.com/elastic/beats/pull/17007/files?#diff-11d6c72e4e8d63c5172a10d494334c75)
- [x] [Quick start](https://beats_17007.docs-preview.app.elstc.co/guide/en/beats/filebeat/master/filebeat-installation-configuration.html) | [Diff](https://github.com/elastic/beats/pull/17007/files?#diff-ff69865c55a33a04334fa556256decf6)
- [x] [Load ingest pipelines](https://beats_17007.docs-preview.app.elstc.co/guide/en/beats/filebeat/master/load-ingest-pipelines.html) | [Diff](https://github.com/elastic/beats/pull/17007/files?#diff-338fcdee5d35526eb1170a02b547a9a1)
- [x] [Override configuration settings](https://beats_17007.docs-preview.app.elstc.co/guide/en/beats/filebeat/master/override-filebeat-config-settings.html) | [Diff](https://github.com/elastic/beats/pull/17007/files?#diff-143c2ba884e0482e7617b486d7288ac4)

**HEARTBEAT**	
- [x] [Quick start](https://beats_17007.docs-preview.app.elstc.co/guide/en/beats/heartbeat/master/heartbeat-installation-configuration.html) | [Diff](https://github.com/elastic/beats/pull/17007/files#diff-90ea3f3cb949a3a9cfc566081b8773cc)

**JOURNALBEAT**	
- [x] [Quick start](https://beats_17007.docs-preview.app.elstc.co/guide/en/beats/journalbeat/master/journalbeat-installation-configuration.html) | [Diff](https://github.com/elastic/beats/pull/17007/files?#diff-89b5952023d2def775d08b24c43ed5d8)

**METRICBEAT**	
- [x] [Quick start](https://beats_17007.docs-preview.app.elstc.co/guide/en/beats/metricbeat/master/metricbeat-installation-configuration.html) | [Diff](https://github.com/elastic/beats/pull/17007/files?#diff-f40ec142e8848b07aebebe590601c6b6)

**PACKETBEAT**	
- [x] [Quick start](https://beats_17007.docs-preview.app.elstc.co/guide/en/beats/packetbeat/master/packetbeat-installation-configuration.html) | [Diff](https://github.com/elastic/beats/pull/17007/files?#diff-e5c83447aba4152c424ae025bb0c16a1)

**WINLOGBEAT**	
- [x] [Quick start](https://beats_17007.docs-preview.app.elstc.co/guide/en/beats/winlogbeat/master/winlogbeat-installation-configuration.html) | [Diff](https://github.com/elastic/beats/pull/17007/files?#diff-8daf5da1e5cd976a03844349ec3ae7d3)

**FUNCTIONBEAT**	
- [x] [Quick start](https://beats_17007.docs-preview.app.elstc.co/guide/en/beats/functionbeat/master/functionbeat-installation-configuration.html) | [Diff](https://github.com/elastic/beats/pull/17007/files?#diff-accdea9577121cac1d096814d9dfeda5)
	
**SHARED	TOPICS**
- [x] [Start Beat](https://beats_17007.docs-preview.app.elstc.co/guide/en/beats/auditbeat/master/auditbeat-starting.html) | [Diff](https://github.com/elastic/beats/pull/17007/files#diff-05747219b7cc3ab4151b96fcf0496afe)
- [x] [Configure Beat](https://beats_17007.docs-preview.app.elstc.co/guide/en/beats/auditbeat/master/configuring-howto-auditbeat.html) | [Diff](https://github.com/elastic/beats/pull/17007/files?#diff-58ce3e22ce9cf86ee160f1fb9ea70bca)
- [x] [Load index templates](https://beats_17007.docs-preview.app.elstc.co/guide/en/beats/auditbeat/master/auditbeat-template.html) | [Diff](https://github.com/elastic/beats/pull/17007/files#diff-e9d35ce95327f554a85f3531a02e9faa)
- [x] [Change index name](https://beats_17007.docs-preview.app.elstc.co/guide/en/beats/auditbeat/master/change-index-name.html) | [Diff](https://github.com/elastic/beats/pull/17007/files#diff-48a130d823fb9268012fa138e2ee32c5)
- [x] [Load Kibana dashboards](https://beats_17007.docs-preview.app.elstc.co/guide/en/beats/filebeat/master/load-kibana-dashboards.html) | [Diff](https://github.com/elastic/beats/pull/17007/files?#diff-6ea4c9030747c37b3ca71715ed0ef717)
	
	
NOTES:

This PR adds a new widget for showing platform-specific content. It is a prototype and up for discussion. The widget is based on some code that @bmorelli25 came up with for an APM prototype plus some CSS changes that I made. Still needs work.

![image](https://user-images.githubusercontent.com/14206422/81452927-1172f300-913d-11ea-87ea-7df6298dd809.png)

elasticmachine commented:
## :broken_heart: Build Failed 

<!-- BUILD BADGES-->
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-17007/detail/PR-17007/12//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-17007/detail/PR-17007/12//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-17007/detail/PR-17007/12//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-17007/detail/PR-17007/12//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://beats_17007.docs-preview.app.elstc.co/diff) [![stats](https://img.shields.io/badge/stats-stats%20-blue)](https://ela.st/observabtl-ci-stats)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats

* Build Cause: [Pull request #17007 updated]
* Start Time: 2020-05-07T01:18:37.981+0000

* Duration: 123 min 1 sec (7380928)


* Commit: 8ac2be5453f00b174444b66e3121cd365d992506



#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 0  |
| Passed       | 7759  |
| Skipped      | 1236 |
| Total        | 8995   |


</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY-->


### Steps errors

  <details><summary>Expand to view the steps failures</summary>
  <p>
  
  * **Name**: Report to Codecov
    * Description:         curl -sSLo codecov https://codecov.io/bash        for i in auditbeat filebeat heartbeat libbeat metricbeat packetbeat winlogbeat journalbeat        do          FILE="${i}/build/coverage/full.cov"          if [ -f "${FILE}" ]; then            bash codecov -f "${FILE}"          fi        done      
    * Result: FAILURE
    
    * Duration: 2 min 25 sec<
    
    * Start Time: 2020-05-07T02:08:39.514+0000
    
  
  </p>
  </details>



<!-- LOG OUTPUT-->
### Log output

<details><summary>Expand to view the last 100 lines of log output</summary>
<p>

```
[2020-05-07T03:20:05.905Z] 	at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:59)
[2020-05-07T03:20:05.905Z] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[2020-05-07T03:20:05.905Z] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[2020-05-07T03:20:05.905Z] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[2020-05-07T03:20:05.905Z] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[2020-05-07T03:20:05.905Z] 	at java.lang.Thread.run(Thread.java:748)
[2020-05-07T03:20:05.905Z] 
[2020-05-07T03:20:06.116Z] Failed in branch Metricbeat x-pack
[2020-05-07T03:20:06.473Z] ERROR: Ignore error temporally
[2020-05-07T03:20:06.473Z] Exception: null
[2020-05-07T03:20:07.114Z] Running on worker-c07ll940dwyl in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007
[2020-05-07T03:21:05.309Z] Body did not finish within grace period; terminating with extreme prejudice
[2020-05-07T03:21:05.732Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats
[2020-05-07T03:21:06.259Z] + find . -type f -name TEST*.xml -path */build/* -delete
[2020-05-07T03:21:06.343Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Lint
[2020-05-07T03:21:06.527Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Elastic-Agent-Mac-OS-X
[2020-05-07T03:21:06.685Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Metricbeat-Mac-OS-X
[2020-05-07T03:21:06.923Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Filebeat-Mac-OS-X
[2020-05-07T03:21:07.139Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Winlogbeat-oss
[2020-05-07T03:21:07.368Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Elastic-Agent-x-pack
[2020-05-07T03:21:07.651Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Journalbeat-oss
[2020-05-07T03:21:07.884Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Dockerlogbeat
[2020-05-07T03:21:08.114Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Functionbeat-x-pack
[2020-05-07T03:21:08.327Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Metricbeat-crosscompile
[2020-05-07T03:21:08.619Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Metricbeat-OSS-Unit-tests
[2020-05-07T03:21:08.832Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Elastic-Agent-x-pack-Windows
[2020-05-07T03:21:09.071Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Heartbeat-oss
[2020-05-07T03:21:09.315Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Auditbeat-Linux
[2020-05-07T03:21:09.542Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Packetbeat-oss
[2020-05-07T03:21:09.767Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Auditbeat-x-pack
[2020-05-07T03:21:10.005Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Generators-Metricbeat-Linux
[2020-05-07T03:21:10.296Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Winlogbeat-Windows-x-pack
[2020-05-07T03:21:10.541Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Filebeat-Windows
[2020-05-07T03:21:10.795Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Libbeat-x-pack
[2020-05-07T03:21:11.016Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Metricbeat-Windows
[2020-05-07T03:21:11.230Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Filebeat-x-pack
[2020-05-07T03:21:11.434Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Auditbeat-crosscompile
[2020-05-07T03:21:11.636Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Filebeat-oss
[2020-05-07T03:21:11.828Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Heartbeat-Mac-OS-X
[2020-05-07T03:21:12.063Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Functionbeat-Mac-OS-X-x-pack
[2020-05-07T03:21:12.302Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Auditbeat-Mac-OS-X
[2020-05-07T03:21:12.500Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Libbeat-oss
[2020-05-07T03:21:12.734Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Metricbeat-OSS-Integration-tests
[2020-05-07T03:21:12.930Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Metricbeat-Python-integration-tests
[2020-05-07T03:21:13.243Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Winlogbeat-Windows
[2020-05-07T03:21:13.507Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Libbeat-crosscompile
[2020-05-07T03:21:13.676Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Heartbeat-Windows
[2020-05-07T03:21:13.941Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Libbeat-stress-tests
[2020-05-07T03:21:14.174Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Auditbeat-Windows
[2020-05-07T03:21:14.398Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Generators-Beat-Linux
[2020-05-07T03:21:14.635Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Functionbeat-Windows
[2020-05-07T03:21:14.840Z] Running in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Metricbeat-x-pack
[2020-05-07T03:21:15.489Z] + cat
[2020-05-07T03:21:15.489Z] + /usr/local/bin/runbld ./runbld-script
[2020-05-07T03:21:15.489Z] Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8
[2020-05-07T03:21:21.320Z] runbld>>> runbld started
[2020-05-07T03:21:21.320Z] runbld>>> 1.6.11/a66728ff8f4356963772e6e6d2069392fa06acbe
[2020-05-07T03:21:23.617Z] runbld>>> The following profiles matched the job 'Beats/beats-beats-mbp/PR-17007' in order of occurrence in the config (last value wins).
[2020-05-07T03:21:24.743Z] runbld>>> Debug logging enabled.
[2020-05-07T03:21:24.743Z] runbld>>> Storing result
[2020-05-07T03:21:24.743Z] runbld>>> Store result: created {:total 2, :successful 2, :failed 0} 1
[2020-05-07T03:21:24.743Z] runbld>>> BUILD: https://c150076387b5421f9154dfbf536e5c60.us-west1.gcp.cloud.es.io:9243/build-1587637540455/t/20200507032124-4679AC24
[2020-05-07T03:21:24.743Z] runbld>>> Adding system facts.
[2020-05-07T03:21:25.891Z] runbld>>> Adding vcs info for the latest commit:  970254b09da8bcab45c9d76221b9b5ed66c4cccc
[2020-05-07T03:21:25.891Z] runbld>>> >>>>>>>>>>>> SCRIPT EXECUTION BEGIN >>>>>>>>>>>>
[2020-05-07T03:21:25.891Z] runbld>>> Adding /usr/lib/jvm/java-8-openjdk-amd64/bin to the path.
[2020-05-07T03:21:25.891Z] + echo 'Processing JUnit reports with runbld...'
[2020-05-07T03:21:25.891Z] Processing JUnit reports with runbld...
[2020-05-07T03:21:26.578Z] runbld>>> <<<<<<<<<<<< SCRIPT EXECUTION END <<<<<<<<<<<<
[2020-05-07T03:21:26.578Z] runbld>>> DURATION: 15ms
[2020-05-07T03:21:26.578Z] runbld>>> STDOUT: 40 bytes
[2020-05-07T03:21:26.578Z] runbld>>> STDERR: 49 bytes
[2020-05-07T03:21:26.578Z] runbld>>> WRAPPED PROCESS: SUCCESS (0)
[2020-05-07T03:21:26.578Z] runbld>>> Searching for build metadata in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats
[2020-05-07T03:21:27.725Z] runbld>>> Storing build metadata: 
[2020-05-07T03:21:27.725Z] runbld>>> Adding test report.
[2020-05-07T03:21:27.725Z] runbld>>> Searching for junit test output files with the pattern: TEST-.*\.xml$ in: /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats
[2020-05-07T03:21:28.822Z] runbld>>> Found 103 test output files
[2020-05-07T03:21:29.509Z] runbld>>> No testsuite node found in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Metricbeat-x-pack/x-pack/metricbeat/build/TEST-go-integration-openmetrics.xml
[2020-05-07T03:21:29.509Z] runbld>>> No testsuite node found in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Metricbeat-x-pack/x-pack/metricbeat/build/TEST-go-integration-istio.xml
[2020-05-07T03:21:29.509Z] runbld>>> No testsuite node found in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Metricbeat-x-pack/x-pack/metricbeat/build/TEST-go-integration-iis.xml
[2020-05-07T03:21:29.509Z] runbld>>> No testsuite node found in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Metricbeat-x-pack/x-pack/metricbeat/build/TEST-go-integration-activemq.xml
[2020-05-07T03:21:29.509Z] runbld>>> No testsuite node found in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Metricbeat-x-pack/x-pack/metricbeat/build/TEST-go-integration-tomcat.xml
[2020-05-07T03:21:29.509Z] runbld>>> No testsuite node found in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Metricbeat-x-pack/x-pack/metricbeat/build/TEST-go-integration-cloudfoundry.xml
[2020-05-07T03:21:30.210Z] runbld>>> No testsuite node found in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Metricbeat-OSS-Integration-tests/metricbeat/build/TEST-go-integration-windows.xml
[2020-05-07T03:21:30.210Z] runbld>>> No testsuite node found in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007/src/github.com/elastic/beats/Metricbeat-OSS-Integration-tests/metricbeat/build/TEST-go-integration-graphite.xml
[2020-05-07T03:21:30.884Z] runbld>>> Test output logs contained: Errors: 0 Failures: 0 Tests: 8845 Skipped: 1020
[2020-05-07T03:21:30.884Z] runbld>>> Storing result
[2020-05-07T03:21:30.884Z] runbld>>> FAILURES: 0
[2020-05-07T03:21:31.195Z] runbld>>> Store result: updated {:total 2, :successful 2, :failed 0} 2
[2020-05-07T03:21:31.195Z] runbld>>> BUILD: https://c150076387b5421f9154dfbf536e5c60.us-west1.gcp.cloud.es.io:9243/build-1587637540455/t/20200507032124-4679AC24
[2020-05-07T03:21:31.195Z] runbld>>> Email notification disabled by environment variable.
[2020-05-07T03:21:31.195Z] runbld>>> Slack notification disabled by environment variable.
[2020-05-07T03:21:37.593Z] Running on Jenkins in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007
[2020-05-07T03:21:37.829Z] [INFO] getVaultSecret: Getting secrets
[2020-05-07T03:21:37.966Z] Masking supported pattern matches of $VAULT_ADDR or $VAULT_ROLE_ID or $VAULT_SECRET_ID
[2020-05-07T03:21:39.360Z] + chmod 755 generate-build-data.sh
[2020-05-07T03:21:39.361Z] + ./generate-build-data.sh https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats-beats-mbp/PR-17007/ https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats-beats-mbp/PR-17007/runs/12 ABORTED 7380928
[2020-05-07T03:21:39.911Z] INFO: curl https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats-beats-mbp/PR-17007/runs/12/steps/?limit=10000 -o steps-info.json
[2020-05-07T03:21:42.395Z] INFO: curl https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats-beats-mbp/PR-17007/runs/12/tests/?status=FAILED -o tests-errors.json

```

</p>
</details>

bmorelli25 commented:
This is fantastic! You've drastically improved my very simple prototype, and turned it into something that we can actually use!

I pushed some CSS changes. A few small changes, and one big change that removes the red border highlighting.

Unfortunately, I was unable to make tab changes apply to all tab boxes on the page. It appears to be a bit beyond my skill level.

bmorelli25 commented on libbeat/docs/tab-widgets/install.asciidoc:
```diff
@@ -0,0 +1,129 @@
+----------------------------------------------------------------------
+
+NOTE: If script execution is disabled on your system, you need to set the
+execution policy for the current session to allow the sc      ript to run. For
```
```suggestion
execution policy for the current session to allow the script to run. For
```

dedemorton replied:
whoopsie.

bmorelli25 commented on libbeat/docs/tab-widgets/stuff.asciidoc:
```diff
@@ -2,7 +2,7 @@
 <style>
 .tabs {
-  width: 40em;
+  width: 100%;
```
So much better!

bmorelli25 commented on libbeat/docs/tab-widgets/stuff.asciidoc:
```diff
@@ -68,35 +65,19 @@
   position: relative;
   z-index: 2;
-  padding: 0.5em 0.5em 0.7em;
+  padding: 1em;
```
I increased the padding inside the tab panel. Everything felt a little too compact to me before. Feel free to revert this if you disagree.

bmorelli25 commented:
I wonder if we could get a front-end dev to look at the CSS/HTML/JS. I'm sure there are still a lot of improvements that can be made, and they might even be able to help us sync the tabs.

bmorelli25 commented on libbeat/docs/tab-widgets/enable-modules-widget.asciidoc:
```diff
@@ -0,0 +1,114 @@
+</div>
+++++
+
+include::stuff.asciidoc[]
```
What about just including this once in `index.asciidoc`?

dedemorton replied:
I think we should add it the stylesheet for all the guides. Just parking it in an asciidoc file for now. I thought the css would need to be included for each html page, though I didn't test the index as you suggest.

dedemorton commented:
Adding a to-do list because there are a lot of moving parts to track:


To-do list:

- [ ] Make the getting started (GS) guide a single topic that describes the easiest getting started experience (for Filebeat, Auditbeat, and Metricbeat, this should be modules).
  - [ ] Packetbeat
  - [ ] Filebeat
  - [ ] Winlogbeat
  - [ ] Metricbeat
  - [ ] Heartbeat
  - [ ] Auditbeat
  - [ ] Functionbeat
  - [ ] Journalbeat
- [ ] Move content that doesn't belong in the GS the how-to section:
  - [ ] Configure the input manually
  - [ ] Changing index name
  - [ ] Loading index templates
- [ ] Simplify conditional coding and content reuse, where feasible (focus on changed files)
- [ ] Add widget for displaying multi-platform commands
- [ ] Set up redirects:
  - [ ] filebeat-configuration.html to configuration-filebeat-options.html
- [ ] Delete extraneous files
  - [ ] shared/download-and-install.asciidoc
  - [ ] step-configure-credentials.asciidoc
  - [ ] step-configure-kibana-endpoint.asciidoc
  - [ ] step-condigure-output.asciidoc
- [ ] Consolidate duplicated content:
  - [ ] In Filebeat docs, delete the quick start (since the new getting started will cover modules) and move any remaining (non-GS content) to the topic about [running modules](https://www.elastic.co/guide/en/beats/filebeat/current/configuration-filebeat-modules.html).

elasticmachine commented:
## :green_heart: Build Succeeded


<!-- BUILD BADGES-->
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-17007/detail/PR-17007/30//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-17007/detail/PR-17007/30//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-17007/detail/PR-17007/30//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-17007/detail/PR-17007/30//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://beats_17007.docs-preview.app.elstc.co/diff)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats

* Build Cause: `[Pull request #17007 updated]`

* Start Time: 2020-05-22T03:37:50.864+0000

* Duration: 75 min 20 sec




#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 0  |
| Passed       | 9214  |
| Skipped      | 1542 |
| Total        | 10756   |


</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY-->

elasticmachine commented:
## :green_heart: Build Succeeded


<!-- BUILD BADGES-->
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-17007/detail/PR-17007/2//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-17007/detail/PR-17007/2//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-17007/detail/PR-17007/2//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-17007/detail/PR-17007/2//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://beats_17007.docs-preview.app.elstc.co/diff)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats

* Build Cause: `[Pull request #17007 updated]`

* Start Time: 2020-05-28T03:46:35.741+0000

* Duration: 88 min 24 sec




#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 0  |
| Passed       | 9242  |
| Skipped      | 1542 |
| Total        | 10784   |


</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY-->


### Steps errors

  <details><summary>Expand to view the steps failures</summary>
  <p>
  
  * **Name**: `Install docker-compose 1.21.0`
    * Description: .ci/scripts/install-docker-compose.sh
    
    * Duration: 1 min 32 sec
    
    * Start Time: 2020-05-28T04:10:59.889+0000
    
    * [log](https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats-beats-mbp/pipelines/PR-17007/runs/2/steps/1467/log)
    
  
  * **Name**: `Report to Codecov`
    * Description:         curl -sSLo codecov https://codecov.io/bash        for i in auditbeat filebeat heartbeat libbeat metricbeat packetbeat winlogbeat journalbeat        do          FILE="${i}/build/coverage/full.cov"          if [ -f "${FILE}" ]; then            bash codecov -f "${FILE}"          fi        done      
    
    * Duration: 2 min 22 sec
    
    * Start Time: 2020-05-28T04:27:10.455+0000
    
    * [log](https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats-beats-mbp/pipelines/PR-17007/runs/2/steps/2644/log)
    
  
  </p>
  </details>

elasticmachine commented:
Pinging @elastic/integrations (Team:Integrations)

elasticmachine commented:
Pinging @elastic/siem (Team:SIEM)

elasticmachine commented:
Pinging @elastic/uptime (Team:Uptime)

bmorelli25 commented on heartbeat/docs/getting-started.asciidoc:
```diff
@@ -1,184 +1,73 @@
-== Get started with {beatname_uc}
+[id="{beatname_lc}-installation-configuration"]
+== {beatname_uc} quick start: installation and configuration
 
 ++++
-<titleabbrev>Get started</titleabbrev>
+<titleabbrev>Quick start: installation and configuration</titleabbrev>
```
Why "Installation and configuration" instead of "Install and configure"?

dedemorton replied:
I checked with Anne B, and those terms rank very high in google searches.

bmorelli25 commented on heartbeat/docs/getting-started.asciidoc:
```diff
@@ -1,184 +1,73 @@
 
-[[heartbeat-installation]]
+[role="screenshot"]
+image::./images/heartbeat-statistics.png[{beatname_uc} HTTP monitoring dashboard]
```
What about showing an uptime app screenshot instead?

dedemorton replied:
IMO showing the app screenshot *instead* of the dashboards muddies the waters a bit for pure OSS users. Also, Observability users will still need to use the finer-grained dashboards to see stuff that isn't visible in the apps.

As a follow-on PR, I'd like to make the "What's next" section a bit more graphical. Maybe we can also add a section (flagged clearly as X-Pack) that showcases the app.  It is a little hard to keep those screen captures up-to-date, though, especially since the team that makes the UI changes won't be updating the Beats docs. WDYT?

bmorelli25 replied:
> As a follow-on PR, I'd like to make the "What's next" section a bit more graphical. Maybe we can also add a section (flagged clearly as X-Pack) that showcases the app.

I like this idea.

> It is a little hard to keep those screen captures up-to-date, though, especially since the team that makes the UI changes won't be updating the Beats docs. WDYT?

Fair point. With the addition of a new observability docs repo though, I think it'd be worthwhile to have one screenshot of each app that we can single source and update at least every other release.

bmorelli25 commented on heartbeat/docs/getting-started.asciidoc:
```diff
@@ -1,184 +1,73 @@
+[float]
+=== Before you begin
+
+You need {es} for storing and searching your data, and {kib} for visualizing and
+managing it.
+
+include::{libbeat-dir}/tab-widgets/spinup-stack-widget.asciidoc[]
```
Side note: It would be awesome if these links could open in a new window

dedemorton replied:
Totally agree, though I think this would require a docs tooling change. We wouldn't want to change the behavior of Beats links without doing the same across the library. Maybe open a request in the docs repo? It's possible there's one already open. :-P

bmorelli25 commented on heartbeat/docs/getting-started.asciidoc:
```diff
@@ -1,184 +1,73 @@
-https://www.elastic.co/downloads/beats/heartbeat[downloads page].
+[float]
+[[set-connection]]
+=== Step 2: Set the connection information
```
What about "Step 2: Connect to the Elastic Stack"

dedemorton replied:
fixed.

bmorelli25 commented on heartbeat/docs/getting-started.asciidoc:
```diff
@@ -198,88 +87,73 @@ was started. Heartbeat adds the `@every` keyword to the syntax provided by the
-By default, Windows log files are stored in +C:\ProgramData\heartbeat\Logs+.
+{beatname_uc} comes with pre-built {kib} dashboards and UIs for visualizing the
+status of your services. The dashboards are available in the
+https://github.com/elastic/uptime-contrib[uptime-contrib] github repository.
```
nit: GitHub
```suggestion
https://github.com/elastic/uptime-contrib[uptime-contrib] GitHub repository.
```

bmorelli25 commented on libbeat/docs/tab-widgets/start.asciidoc:
```diff
@@ -0,0 +1,148 @@
+----------------------------------------------------------------------
+
+// tag::initd-note[]
+NOTE: If you use an init.d script to start {beatname_uc}, you can't specify command
```
```suggestion
NOTE: If you use an `init.d` script to start {beatname_uc}, you can't specify command
```

bmorelli25 commented on heartbeat/docs/getting-started.asciidoc:
```diff
@@ -198,88 +87,73 @@ was started. Heartbeat adds the `@every` keyword to the syntax provided by the
+The `setup` command loads the recommended index template for writing to {es}.
+It does not install {beatname_uc} dashboards.  Heartbeat dashboards and 
+installation steps are available in the
+https://github.com/elastic/uptime-contrib[uptime-contrib] github repository.
```
```suggestion
https://github.com/elastic/uptime-contrib[uptime-contrib] GitHub repository.
```

bmorelli25 commented on heartbeat/docs/getting-started.asciidoc:
```diff
@@ -198,88 +87,73 @@ was started. Heartbeat adds the `@every` keyword to the syntax provided by the
-events to your defined output.
+If you loaded the dashboards earlier, open them now.
 
-[[view-kibana-dashboards]]
-=== Step 6: View the sample Kibana dashboards
+include::{libbeat-dir}/shared/opendashboards.asciidoc[tag=open-dashboards]
```
> If you loaded the dashboards earlier, open them now.
>
> To open the dashboards:

Could we simplify to `If you loaded the dashboards earlier, open them now:`

bmorelli25 replied:
The more I think about this, the less I like my suggestion

dedemorton replied:
I think I like you suggestions, but will need to look in context.

dedemorton replied:
Hmm...Heartbeat is an outlier from the other Beats because the dashboards are not loaded when you run the setup command. (That's why the style here is a bit different.) 

I'm going to leave as-is for now because I can't think of a way to get around this problem without adding conditional coding, which I don't want to do.

bmorelli25 replied:
Sounds good to me

bmorelli25 commented on libbeat/docs/shared/opendashboards.asciidoc:
```diff
@@ -0,0 +1,28 @@
+sure the predefined +{beatname_lc}-*+ index pattern is selected.
++
+--
+TIP: If you don’t see data in {kib}, try changing the date range to a larger
```
Officially, it's called the `time filter`.

```suggestion
TIP: If you don’t see data in {kib}, try changing the time filter to a larger
```

bmorelli25 commented on libbeat/docs/howto/load-index-templates.asciidoc:
```diff
@@ -141,7 +107,7 @@ ifdef::no-output-logstash[]
+include::{libbeat-dir}/shared-note-sudo.asciidoc[]
 endif::requires-sudo[]
 
 ifdef::deb_os,rpm_os[]
```
A project for another day, but I can't wait to see the tab widget in use here as well!

dedemorton replied:
Yes! My priority is getting the quick start to show the tabs. Didn't want to do the other topics until after this PR is merged and we have the final source code for the tabbed panel in place.

bmorelli25 replied:
💯

dedemorton commented on auditbeat/docs/getting-started.asciidoc:
```diff
@@ -1,182 +1,70 @@
-* <<{beatname_lc}-starting>>
-* <<view-kibana-dashboards>>
-* <<setup-repositories>>
+* install {beatname_uc} on each system you want to monitor
+* specify the location of your audit data
+* parse log data into fields and send it to {es}
+* visualize the log data in {kib}
```
@bmorelli25 I didn't think the links were necessary now that everything is on one page. It's pretty short, and we want users to follow in order.

Why no caps? Maybe I was in a weird mood? I wonder if caps are needed when the list item isn't a complete sentence.

bmorelli25 replied:
> Maybe I was in a weird mood?

😂

elasticmachine commented:
## :grey_exclamation: Build Aborted
  
  > There is a new build on-going so the previous on-going builds have been aborted.
  


<!-- BUILD BADGES-->
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-17007/detail/PR-17007/13//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-17007/detail/PR-17007/13//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-17007/detail/PR-17007/13//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-17007/detail/PR-17007/13//artifacts) 

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats

* Build Cause: `[Pull request #17007 updated]`

* Reason: `Aborted from #14`

* Start Time: 2020-06-04T17:53:25.075+0000

* Duration: 3 min 59 sec


* Commit: 4a8a003fb7df256f80991684e8a736ca0340da44




</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY-->




<!-- LOG OUTPUT-->
### Log output

<details><summary>Expand to view the last 100 lines of log output</summary>
<p>

```
[2020-06-04T17:56:21.294Z] Stage "Elastic Agent x-pack" skipped due to earlier failure(s)
[2020-06-04T17:56:21.295Z] Stage "Elastic Agent x-pack Windows" skipped due to earlier failure(s)
[2020-06-04T17:56:21.296Z] Stage "Elastic Agent Mac OS X" skipped due to earlier failure(s)
[2020-06-04T17:56:21.297Z] Stage "Filebeat oss" skipped due to earlier failure(s)
[2020-06-04T17:56:21.298Z] Stage "Filebeat x-pack" skipped due to earlier failure(s)
[2020-06-04T17:56:21.299Z] Stage "Filebeat Mac OS X" skipped due to earlier failure(s)
[2020-06-04T17:56:21.299Z] Stage "Filebeat x-pack Mac OS X" skipped due to earlier failure(s)
[2020-06-04T17:56:21.300Z] Stage "Filebeat Windows" skipped due to earlier failure(s)
[2020-06-04T17:56:21.301Z] Stage "Filebeat x-pack Windows" skipped due to earlier failure(s)
[2020-06-04T17:56:21.302Z] Stage "Heartbeat" skipped due to earlier failure(s)
[2020-06-04T17:56:21.302Z] Stage "Auditbeat oss Linux" skipped due to earlier failure(s)
[2020-06-04T17:56:21.303Z] Stage "Auditbeat crosscompile" skipped due to earlier failure(s)
[2020-06-04T17:56:21.304Z] Stage "Auditbeat oss Mac OS X" skipped due to earlier failure(s)
[2020-06-04T17:56:21.305Z] Stage "Auditbeat oss Windows" skipped due to earlier failure(s)
[2020-06-04T17:56:21.306Z] Stage "Auditbeat x-pack" skipped due to earlier failure(s)
[2020-06-04T17:56:21.307Z] Stage "Auditbeat x-pack Mac OS X" skipped due to earlier failure(s)
[2020-06-04T17:56:21.307Z] Stage "Auditbeat x-pack Windows" skipped due to earlier failure(s)
[2020-06-04T17:56:21.308Z] Stage "Libbeat" skipped due to earlier failure(s)
[2020-06-04T17:56:21.309Z] Stage "Libbeat x-pack" skipped due to earlier failure(s)
[2020-06-04T17:56:21.309Z] Stage "Metricbeat OSS Unit tests" skipped due to earlier failure(s)
[2020-06-04T17:56:21.310Z] Stage "Metricbeat OSS Integration tests" skipped due to earlier failure(s)
[2020-06-04T17:56:21.311Z] Stage "Metricbeat Python integration tests" skipped due to earlier failure(s)
[2020-06-04T17:56:21.312Z] Stage "Metricbeat x-pack" skipped due to earlier failure(s)
[2020-06-04T17:56:21.312Z] Stage "Metricbeat crosscompile" skipped due to earlier failure(s)
[2020-06-04T17:56:21.313Z] Stage "Metricbeat Mac OS X" skipped due to earlier failure(s)
[2020-06-04T17:56:21.314Z] Stage "Metricbeat x-pack Mac OS X" skipped due to earlier failure(s)
[2020-06-04T17:56:21.315Z] Stage "Metricbeat Windows" skipped due to earlier failure(s)
[2020-06-04T17:56:21.316Z] Stage "Metricbeat x-pack Windows" skipped due to earlier failure(s)
[2020-06-04T17:56:21.317Z] Stage "Packetbeat" skipped due to earlier failure(s)
[2020-06-04T17:56:21.318Z] Stage "dockerlogbeat" skipped due to earlier failure(s)
[2020-06-04T17:56:21.318Z] Stage "Winlogbeat" skipped due to earlier failure(s)
[2020-06-04T17:56:21.319Z] Stage "Winlogbeat Windows x-pack" skipped due to earlier failure(s)
[2020-06-04T17:56:21.320Z] Stage "Functionbeat" skipped due to earlier failure(s)
[2020-06-04T17:56:21.321Z] Stage "Journalbeat" skipped due to earlier failure(s)
[2020-06-04T17:56:21.321Z] Stage "Generators" skipped due to earlier failure(s)
[2020-06-04T17:56:21.322Z] Stage "Kubernetes" skipped due to earlier failure(s)
[2020-06-04T17:56:21.413Z] Stage "Heartbeat" skipped due to earlier failure(s)
[2020-06-04T17:56:21.415Z] Stage "Libbeat" skipped due to earlier failure(s)
[2020-06-04T17:56:21.416Z] Stage "Metricbeat x-pack" skipped due to earlier failure(s)
[2020-06-04T17:56:21.417Z] Stage "Packetbeat" skipped due to earlier failure(s)
[2020-06-04T17:56:21.417Z] Stage "dockerlogbeat" skipped due to earlier failure(s)
[2020-06-04T17:56:21.418Z] Stage "Winlogbeat" skipped due to earlier failure(s)
[2020-06-04T17:56:21.419Z] Stage "Functionbeat" skipped due to earlier failure(s)
[2020-06-04T17:56:21.420Z] Stage "Journalbeat" skipped due to earlier failure(s)
[2020-06-04T17:56:21.421Z] Stage "Generators" skipped due to earlier failure(s)
[2020-06-04T17:56:21.939Z] Failed in branch Elastic Agent x-pack
[2020-06-04T17:56:21.940Z] Failed in branch Elastic Agent x-pack Windows
[2020-06-04T17:56:21.941Z] Failed in branch Elastic Agent Mac OS X
[2020-06-04T17:56:21.942Z] Failed in branch Filebeat oss
[2020-06-04T17:56:21.942Z] Failed in branch Filebeat x-pack
[2020-06-04T17:56:21.943Z] Failed in branch Filebeat Mac OS X
[2020-06-04T17:56:21.944Z] Failed in branch Filebeat x-pack Mac OS X
[2020-06-04T17:56:21.944Z] Failed in branch Filebeat Windows
[2020-06-04T17:56:21.945Z] Failed in branch Filebeat x-pack Windows
[2020-06-04T17:56:21.946Z] Failed in branch Auditbeat oss Linux
[2020-06-04T17:56:21.946Z] Failed in branch Auditbeat crosscompile
[2020-06-04T17:56:21.947Z] Failed in branch Auditbeat oss Mac OS X
[2020-06-04T17:56:21.948Z] Failed in branch Auditbeat oss Windows
[2020-06-04T17:56:21.948Z] Failed in branch Auditbeat x-pack
[2020-06-04T17:56:21.949Z] Failed in branch Auditbeat x-pack Mac OS X
[2020-06-04T17:56:21.949Z] Failed in branch Auditbeat x-pack Windows
[2020-06-04T17:56:21.950Z] Failed in branch Libbeat x-pack
[2020-06-04T17:56:21.951Z] Failed in branch Metricbeat OSS Unit tests
[2020-06-04T17:56:21.951Z] Failed in branch Metricbeat OSS Integration tests
[2020-06-04T17:56:21.952Z] Failed in branch Metricbeat Python integration tests
[2020-06-04T17:56:21.953Z] Failed in branch Metricbeat crosscompile
[2020-06-04T17:56:21.954Z] Failed in branch Metricbeat Mac OS X
[2020-06-04T17:56:21.954Z] Failed in branch Metricbeat x-pack Mac OS X
[2020-06-04T17:56:21.955Z] Failed in branch Metricbeat Windows
[2020-06-04T17:56:21.955Z] Failed in branch Metricbeat x-pack Windows
[2020-06-04T17:56:21.956Z] Failed in branch Winlogbeat Windows x-pack
[2020-06-04T17:56:21.957Z] Failed in branch Kubernetes
[2020-06-04T17:56:22.209Z] Stage "Heartbeat" skipped due to earlier failure(s)
[2020-06-04T17:56:22.211Z] Stage "Libbeat" skipped due to earlier failure(s)
[2020-06-04T17:56:22.212Z] Stage "Metricbeat x-pack" skipped due to earlier failure(s)
[2020-06-04T17:56:22.213Z] Stage "Winlogbeat" skipped due to earlier failure(s)
[2020-06-04T17:56:22.214Z] Stage "Functionbeat" skipped due to earlier failure(s)
[2020-06-04T17:56:22.215Z] Stage "Generators" skipped due to earlier failure(s)
[2020-06-04T17:56:22.272Z] Failed in branch Packetbeat
[2020-06-04T17:56:22.273Z] Failed in branch dockerlogbeat
[2020-06-04T17:56:22.274Z] Failed in branch Journalbeat
[2020-06-04T17:56:22.463Z] Stage "Heartbeat" skipped due to earlier failure(s)
[2020-06-04T17:56:22.464Z] Stage "Libbeat" skipped due to earlier failure(s)
[2020-06-04T17:56:22.465Z] Stage "Functionbeat" skipped due to earlier failure(s)
[2020-06-04T17:56:22.466Z] Stage "Generators" skipped due to earlier failure(s)
[2020-06-04T17:56:22.502Z] Failed in branch Metricbeat x-pack
[2020-06-04T17:56:22.503Z] Failed in branch Winlogbeat
[2020-06-04T17:56:22.667Z] Failed in branch Heartbeat
[2020-06-04T17:56:22.668Z] Failed in branch Libbeat
[2020-06-04T17:56:22.669Z] Failed in branch Functionbeat
[2020-06-04T17:56:22.670Z] Stage "Generators" skipped due to earlier failure(s)
[2020-06-04T17:56:22.741Z] Failed in branch Generators
[2020-06-04T17:56:23.349Z] Running on Jenkins in /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-17007
[2020-06-04T17:56:23.591Z] [INFO] getVaultSecret: Getting secrets
[2020-06-04T17:56:23.646Z] Masking supported pattern matches of $VAULT_ADDR or $VAULT_ROLE_ID or $VAULT_SECRET_ID
[2020-06-04T17:56:24.459Z] + chmod 755 generate-build-data.sh
[2020-06-04T17:56:24.459Z] + ./generate-build-data.sh https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats-beats-mbp/PR-17007/ https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats-beats-mbp/PR-17007/runs/13 ABORTED 179123
[2020-06-04T17:56:24.459Z] INFO: curl https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats-beats-mbp/PR-17007/runs/13/steps/?limit=10000 -o steps-info.json
[2020-06-04T17:56:24.709Z] INFO: curl https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats-beats-mbp/PR-17007/runs/13/tests/?status=FAILED -o tests-errors.json
[2020-06-04T17:56:24.709Z] INFO: curl https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats-beats-mbp/PR-17007/runs/13/log/ -o pipeline-log.txt

```

</p>
</details>

dedemorton commented:
@jsoriano Thanks for taking a look!! I'll include my responses here inline:

* Links to install beats in the getting started for the elastic stack go to pages that only contain a link to the quick start guides. Could we link directly to the quick start guides?
> Yes, absolutely. That'll be a separate PR in stack-docs. The "deleted" pages are temporary until we get all the links updated and have redirects in place. 
* Should the new tabs widget be used in these pages too?
Loading templates: https://beats_17007.docs-preview.app.elstc.co/guide/en/beats/packetbeat/master/packetbeat-template.html
Loading kibana dashboards: https://beats_17007.docs-preview.app.elstc.co/guide/en/beats/filebeat/master/load-kibana-dashboards.html
> Yes! That's in the plan, but it'll be a follow-up PR once we have the tabs working exactly like we want them to.
* Would it make sense to add an index in pages with many sections? Now that all setup steps are in one page, it could make sense to have a quick way to to go to specific sections, for example in functionbeat page to go directly to the section about google cloud.
> I think you're describing a jump list that takes you to the sections, no? I could turn the bulleted list at the top into a jump list, but that sort of duplicates what we already have in the "On this page" area (upper right corner). I've felt for a long time that the "On this page" area isn't prominent enough. I'm hoping that the site redesign will fix this problem. I'd like to see the actual TOC on the left side, and the "on this page" list on the right, with links that get highlighted as you scroll through the page.
* Btw, also in the functionbeat page, there are examples and instructions for different cloud providers in multiple places. Would it make sense to use the tabbed widget also there? Or maybe have a different quick start guide for each cloud provider?
> I actually started to build this widget, but then got into a problem where I would have to embed a tabbed panel in another tab. I decided that it was OK to show examples of both for now. Your idea of providing separate quick starts intrigues me. I guess the deciding factor would be whether it's likely that users will want to collect data from multiple cloud providers, in which case, having it all in one file might make sense. I just want to make sure we don't end up forcing users to read through the same setup content over and over again if they're likely to be installing a beat to monitor multiple services.
* I think we should explain how to change the index name with ILM in this page, this is a frequent question: https://beats_17007.docs-preview.app.elstc.co/guide/en/beats/auditbeat/master/change-index-name.html But probably better to do it in a different PR.
> Totally agree. I wanted to minimize the actual content changes in files that I moved though. I think the content I've moved out of the getting started guide definitely needs more work to stand alone as topics. That'll be a follow up PR.

Regarding the use of sudo, I haven't seen anything specially wrong. There are some cases where the commands would work without sudo, like here, starting Metricbeat when installing from tar in Linux or Mac would work, but sudo is needed in other beats like Auditbeat. I think that better we leave this with sudo to avoid problems with the beats that need it.
Do you have doubts in the use of sudo in other places? I guess than in any case this could be reviewed in a future PR.
> There are some places where the docs don't match the steps in the add data pages. Let's review this in a future PR.

Thanks again! I really appreciate your input.

benskelker commented:
> I wonder if we could get a front-end dev to look at the CSS/HTML/JS. I'm sure there are still a lot of improvements that can be made, and they might even be able to help us sync the tabs.

To sync the tabs, you could do something like this with the event lister `changeTabs` function:

~~~~
function changeTabs(e) {
  const target = e.target;
  
  var tabButtons = document.querySelectorAll('[role="tab"]');
  for (var i = 0; i < tabButtons.length; i++) {
    
    parent = tabButtons[i].parentNode
    grandparent = parent.parentNode
    
    if (target.innerText == tabButtons[i].innerText) {
      // Remove all current selected tabs
      parent
        .querySelectorAll('[aria-selected="true"]')
        .forEach(t => t.setAttribute("aria-selected", false));

      // Set this tab as selected
      tabButtons[i].setAttribute("aria-selected", true);

      // Hide all tab panels
      grandparent
        .querySelectorAll('[role="tabpanel"]')
        .forEach(p => p.setAttribute("hidden", true));

      // Show the selected panel
      grandparent.parentNode
        .querySelector(`#${tabButtons[i].getAttribute("aria-controls")}`)
        .removeAttribute("hidden");
    }
  }
}
~~~~

Although, definitely think someone from dev should take a look. Also, might be better not to repeat the JS/CSS code multiple times on the same page (remove the stuff.asciidoc include from the widget files and add an include to the doc files whenever tabs are used). Not sure if it really matters.

dedemorton commented:
@benskelker Thanks! We have someone who is doing a full review of the code, too. Re your comment:

> Also, might be better not to repeat the JS/CSS code multiple times on the same page (remove the stuff.asciidoc include from the widget files and add a stuff.asciidoc include to the doc files when tabs are used).

Don't worry...the asciidoc `include` won't be there in the final implementation. It was just a quick way to get things working. :-)

dedemorton commented:
> @dedemorton @bmorelli25 LGTM My only comment relates to the naming of the titles. Should we use "quick start" in the titles? Some users, especially if they choose to self-manage the stack, may not find the process that quick. And also, the title "Auditbeat quick start: installation and configuration" seems a little lengthy. Maybe to be consistent with the other headings, such as "Set up and run", we could go with "Auditbeat: install and configure" or just "Install and configure"?

Re titles: I'll check google analytics (and look back at my notes from my meeting with Anne B). I came up with that title after meeting with Anne B to discuss the best title + URL to use for SEO. "Get started" didn't rank very highly, so we decided to drop it. I can't remember if "quick start" was necessary. I do think it's good to have some kind of signpost to tell users "start here".

sayden commented on auditbeat/docs/getting-started.asciidoc:
```diff
@@ -189,105 +77,75 @@ auditbeat.modules:
-----------------------------------------------------------------------
-PS C:{backslash}Program Files{backslash}{beatname_uc}> Start-Service {beatname_lc}
-----------------------------------------------------------------------
+//REVIEWERS: Removing cURL test because I think users will see a better view of
```
I guess you don't want to merge this :)

sayden replied:
Ignore this comment if you do :smile:

dedemorton replied:
Thanks! I'll remove all the reviewer notes in the next commit.

sayden commented on metricbeat/docs/getting-started.asciidoc:
```diff
@@ -0,0 +1,164 @@
++
+`-e` is optional and sends output to standard error instead of syslog.
+
+This step loads the recommended index template for writing to {es}
```
I was wondering if it is a good approach to put a link to the index templates here https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-templates.html for users who are unaware of them. Ignore this comment if not :smile:

dedemorton replied:
Good idea! I'll do that.

urso commented on auditbeat/docs/getting-started.asciidoc:
```diff
@@ -1,182 +1,70 @@
 
-ifeval::["{release-state}"!="unreleased"]
+[float]
+[[install]]
```
not sure how asciidoctor handles these anchors. But as we introduce new labels here, would it make sense to introduce a more complete naming schema? e.g. `install` is a quite generic name.

dedemorton replied:
Hmm...that's a good point. In most cases, I just kept the anchor names the same to keep from having to fix links. I think it would actually be pretty useful to have a naming convention. I'm going to defer this for now, but going forward (for new docs) we should come up with a consistent naming pattern for anchors. Naming is especially important in situations where the anchors become the html filename (not here).

dedemorton replied:
Created an issue to track this: https://github.com/elastic/beats/issues/19229

urso commented on auditbeat/docs/getting-started.asciidoc:
```diff
@@ -189,105 +77,75 @@ auditbeat.modules:
+include::{libbeat-dir}/tab-widgets/setup-widget.asciidoc[]
+--
++
+`-e` is optional and sends output to standard error instead of syslog.
```
how about: instead of the configured log output.

urso commented on auditbeat/docs/getting-started.asciidoc:
```diff
@@ -189,105 +77,75 @@ auditbeat.modules:
-----------------------------------------------------------------------
+//REVIEWERS: Removing cURL test because I think users will see a better view of
+//the data simply by going to the Discover app in Kibana. Showing the cURL
+//command is not ideal for cloud users.
```
:+1:

elasticmachine commented:
## :green_heart: Build Succeeded


<!-- BUILD BADGES-->
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-17007/detail/PR-17007/18//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-17007/detail/PR-17007/18//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-17007/detail/PR-17007/18//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats-beats-mbp%2FPR-17007/detail/PR-17007/18//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://beats_17007.docs-preview.app.elstc.co/diff)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats

* Build Cause: `[Pull request #17007 updated]`

* Start Time: 2020-06-16T23:54:15.915+0000

* Duration: 81 min 5 sec




#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 0  |
| Passed       | 9441  |
| Skipped      | 1574 |
| Total        | 11015   |


</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY-->


### Steps errors

  <details><summary>Expand to view the steps failures</summary>
  <p>
  
  * **Name**: `Make -C generator/_templates/beat test`
    * Description: make -C generator/_templates/beat test
    
    * Duration: 7 min 9 sec
    
    * Start Time: 2020-06-17T01:05:16.983+0000
    
    * [log](https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats-beats-mbp/pipelines/PR-17007/runs/18/steps/3571/log)
    
  
  </p>
  </details>




<!--COMMENT_GENERATED_WITH_ID_comment.id-->
